### DISCUSSION TITLE
Can't test with jest, create react and typescript

### DISCUSSION THREAD

**Discussion by lzfrs** [pmndrs/zustand#1736] [Q&A]
Hello there,

I created a project with create-react-app with typescript and started using Zustand
But while testing, even following the documentation, I'm getting the following error:

![image](https://user-images.githubusercontent.com/33661847/230462374-6c284232-3a08-4642-8335-bea3076e313d.png)

Here's the repo:
https://github.com/luizzzdev/zustand-test-error

Already tried everything I've found about this error, but no success :l

Can somebody please help?

**Comment by JacobWeisenburger** [comment]
Perhaps this will help:

```ts
import { createStore, useStore } from 'zustand'

export interface CountStore {
    count: number
    add: () => void
}

export const countStore = createStore<CountStore>( ( set ) => ( {
    count: 0,
    add: () => set( prev => ( {
        ...prev,
        count: prev.count + 1,
    } ) ),
} ) )

export function Count () {
    const { count, add } = useStore( countStore )

    return <div>
        <h1>Count</h1>
        <h2>{count}</h2>
        <button onClick={add}>add</button>
    </div>
}
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Comment by lzfrs** [comment]
No, same error :l
Thanks for the help

**Comment by JacobWeisenburger** [comment]
it seems that zustand doesn't want to work well with `@testing-library/react`

I have run your repo and it renders fine in the browser, it's just the tests that aren't working

**Comment by dbritto-dev** [comment]
Hey folks, could you try again using this guide https://github.com/pmndrs/zustand/blob/main/docs/guides/testing.md? Is something doesn't work just tag me
